### DISCUSSION TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### DISCUSSION THREAD

**Discussion by roughnecks** [ente-io/ente#1474] [Q&A] [answered]
Hi,

me and a friend of mine are setting up local ente servers on our VPSs.
I did it first and everything is working, while he cannot start museum with the following error message.
We're using the same exact setup - compose.yaml, meusem.yaml and credentials.yaml, with a postgresql 15 database on the host system (Debian Stable), rather than the containerized one.

Why on his system it tries to migrate and fails? I don't remember having seen any migration messages when I started my container.
In compose.yaml we have:

`network_mode: "host"`

and in credentials.yaml:

```
db:
    host: localhost
```

```
museum-1  | goroutine 39 [running]:
museum-1  | github.com/golang-migrate/migrate/v4.(*Migrate).lock.func3()
museum-1  |     /go/pkg/mod/github.com/golang-migrate/migrate/v4@v4.12.2/migrate.go:921 +0x29
museum-1  | created by github.com/golang-migrate/migrate/v4.(*Migrate).lock
museum-1  |     /go/pkg/mod/github.com/golang-migrate/migrate/v4@v4.12.2/migrate.go:920 +0x1ea
museum-1  | INFO[0000]main.go:91 main Booting up local server with commit #7be2c66fb69517eb1a2f67b0d94cc72817687df9 
museum-1  | INFO[0000]main.go:753 setupDatabase Setting up db                                
museum-1  | INFO[0000]main.go:760 setupDatabase Connected to DB                              
museum-1  | INFO[0000]main.go:765 setupDatabase Pinged DB                                    
museum-1  | INFO[0000]main.go:774 setupDatabase Loaded migration scripts                     
museum-1  | panic: runtime error: invalid memory address or nil pointer dereference
museum-1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x108a3c9]
```

Thanks

**Comment by mnvr** [comment]
> postgresql 15 

We use postgres 12. While nothing in our SQL is postgres 12 specific to our knowledge, but we haven't tested in any other configuration. So your friend using postgres 15 is likely the reason for the issue they are having.

**Reply by roughnecks** [reply]
But I'm using postgres 15 as well.

**Comment by juli4ndev** [comment, marked as answer]
Hi @roughnecks
I solved this problem with the following commands. (I'm not an expert in postgres security and it may not be the best solution, but in my local environment it was what worked for me.)
```
GRANT ALL PRIVILEGES ON DATABASE ente TO ente;
ALTER DATABASE ente OWNER TO ente;
```

**Reply by roughnecks** [reply]
Thanks, now even my friend is able to run it.
